plugins/@grouparoo/pardot/src/lib/export/destinationMappingOptions.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  DestinationMappingOptionsMethod,
+  DestinationMappingOptionsResponseProperty,
+  DestinationMappingOptionsResponseTypes,
+} from "@grouparoo/core";
+
+import PardotClient from "../client";
+import { connect } from "./../connect";
+
+// Prospect field spec https://developer.pardot.com/kb/object-field-references/#prospect
+const requiredProspectFields: DestinationMappingOptionsResponseProperty[] = [
+  { key: "email", type: "email" },
+];
+
+const defaultProspectFields: DestinationMappingOptionsResponseProperty[] = [
+  { key: "campaign_id", type: "integer" }, // TODO check for validation
+  { key: "salutation", type: "string" },
+  { key: "first_name", type: "string", important: true },
+  { key: "last_name", type: "string", important: true },
+  { key: "password", type: "string" },
+  { key: "company", type: "string" },
+  { key: "prospect_account_id", type: "integer" }, // TODO: check for validation
+  { key: "website", type: "string" },
+  { key: "job_title", type: "string" },
+  { key: "department", type: "string" },
+  { key: "country", type: "string" },
+  { key: "address_one", type: "string" },
+  { key: "address_two", type: "string" },
+  { key: "city", type: "string" },
+  { key: "state", type: "string" },
+  { key: "territory", type: "string" },
+  { key: "zip", type: "string" },
+  { key: "phone", type: "phoneNumber" },
+  { key: "fax", type: "phoneNumber" },
+  { key: "source", type: "string" },
+  { key: "annual_revenue", type: "string" },
+  { key: "employees", type: "string" },
+  { key: "industry", type: "string" },
+  { key: "years_in_business", type: "string" },
+  { key: "comments", type: "string" },
+  { key: "notes", type: "string" },
+  { key: "score", type: "integer" },
+];
+
+export const destinationMappingOptions: DestinationMappingOptionsMethod = async ({
+  appOptions,
+}) => {
+  const client = await connect(appOptions);
+  const known = await getProspectFields(client);
+
+  return {
+    labels: {
+      property: {
+        singular: "Pardot Field",
+        plural: "Pardot Fields",
+      },
+      group: {
+        singular: "Pardot List",
+        plural: "Pardot Lists",
+      },
+    },
+    properties: {
+      known,
+      required: requiredProspectFields,
+      allowOptionalFromProperties: false,
+    },
+  };
+};
+
+const mapTypesToGrouparoo = (
+  pardotType: string
+): DestinationMappingOptionsResponseTypes => {
+  // https://help.salesforce.com/articleView?id=sf.pardot_field_types.htm&type=5
+
+  const map = {
+    Text: "string",
+    Number: "number",
+    Date: "date",
+    Textarea: "string",
+    Dropdown: "any",
+    Hidden: "any",
```

Comment:
I'm not sure about these `any` values. they get us in trouble. is the actual type defined anywhere else? when you made it? Will `"string"` work?

Reply:
Yeah, `string` would work. The reason why I did `any` is because you can set it to integers or strings and it happily accepts them. I _would_ be more comfortable just setting it to a string though...

Reply:
string it is! grouparoo will allow integers to map to a string if they want to map other types.